torch/distributed/fsdp/flat_param.py

```diff
@@ -845,25 +845,52 @@ def _free_low_precision_sharded_param(self):
 
     @torch.no_grad()
     def unshard_grad(self):
+        """
+        Unshards the handle's ``FlatParameter`` 's gradient. If all ranks have
```

Comment:
nit: maybe add a comment that 'unshard_grad' is not used in critical path, only used in summon_full_params(), as it calls all_reduce and may have performance impact